### PR TITLE
[WIP] Kubernetes Marketplace - add subscribe page

### PR DESCRIPTION
**Pull request by galletti94**
Solves [JIRA 875](https://jira.coreos.com/browse/CONSOLE-875)

Adds the subscribe page UI. Creates a CatalogSourceConfig object with the selected items from the dropdowns.

Note: please wait for PR #805 to be merged before thorough review

### CODE REVIEW

File: frontend/public/components/marketplace/kubernetes-marketplace-subscribe.jsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+          id={id}
+          items={items}
+          dropDownClassName="dropdown--full-width"
+          onChange={onChange}
+        />}
+      </div>
+    );
+  }
+}
+class AdminSubscribe extends React.Component {
```
Comment (jeff-phillips-18) [inline frontend/public/components/marketplace/kubernetes-marketplace-subscribe.jsx]: Can we name the file after the class (admin-subscribe.jsx)? It gets confusing when they differ.

File: frontend/public/components/marketplace/kubernetes-marketplace-subscribe.jsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+          />
+          <div style={{ marginTop : '20px' }}>
+            {description}
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+export { AdminSubscribe };
```
Comment (jeff-phillips-18) [inline frontend/public/components/marketplace/kubernetes-marketplace-subscribe.jsx]: Add the propTypes so its a bit easier to know what properties the dialog expects.

File: frontend/public/components/marketplace/kubernetes-marketplace-subscribe.jsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+              Cancel
+            </Button>
+          </div>
+        </div>
+        <div className="co-marketplace-subscribe--description">
+          <CatalogItemHeader
+            className="long-description-test"
+            iconImg={imgUrl}
+            title={name}
+            vendor={<span> {provider}</span>}
```
Comment (jeff-phillips-18) [inline frontend/public/components/marketplace/kubernetes-marketplace-subscribe.jsx]: This should be consistent with the details header:

```suggestion
            vendor={`${version} provided by ${provider}`}
```

File: frontend/public/components/marketplace/kubernetes-marketplace.jsx [hunk 4]
```diff
@@ -113,13 +113,5 @@ export const Marketplace = () => {
...
-  return <React.Fragment>
-    <Helmet>
-      <title>Kubernetes Marketplace</title>
-    </Helmet>
-    <div className="co-catalog">
-      <PageHeading title="Kubernetes Marketplace" />
-      <Marketplace />
-    </div>
-  </React.Fragment>;
+  return <Marketplace />
```
Comment (jeff-phillips-18) [inline frontend/public/components/marketplace/kubernetes-marketplace.jsx]: I believe you can simply rename `Marketplace` to `MarketplacePage` and remove this `MarketplacePage` component. Or even better, converge all three (Marketplace, MarketplacePage, and MarketplaceListPage) into one MarketplacePage component.

**Comment by alecmerdler** [comment]
@galletti94 You probably would be better off making a new branch+PR off `master` (especially since this branch is named `milestone2-demo`!). I will be happy to review once you do.

**Comment by galletti94** [comment]
@alecmerdler good point. I will close this one once I create the new one (probably tomorrow morning)

**Pull request closed without merging**